multiplex.go

```diff
@@ -161,8 +163,8 @@ func (mp *Multiplex) sendMsg(ctx context.Context, header uint64, data []byte) er
 		return nil
 	case <-mp.shutdown:
 		return ErrShutdown
-	case <-ctx.Done():
-		return ctx.Err()
+	case <-done:
+		return errors.New("invalid context")
```

Comment:
? Shouldn't this be a timeout error?

Reply:
Yeah, good point. Is there an existing error we can reuse?

Reply:
Not that I know of. We just need an error that returns true for both `Timeout() bool` and `Temporary() bool`.

Reply:
I made an `errTimeout`, perhaps we want to export this (or reuse an existing error if applicable)

Reply:
ah, we need the `Timeout` and `Temporary` methods too, let me add.

Reply:
done